libdnf/base/goal.cpp

```diff
@@ -905,6 +936,62 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
     }
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GroupJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    comps::GroupQuery group_query(base);
+    comps::GroupQuery group_query_names(group_query);
+    group_query_names.filter_name(spec, sack::QueryCmp::IGLOB);
+    group_query.filter_groupid(spec, sack::QueryCmp::IGLOB);
+    group_query |= group_query_names;
+    if (group_query.empty()) {
+        auto problem = transaction.p_impl->report_not_found(GoalAction::INSTALL, spec, GoalJobSettings(), strict);
+        if (strict) {
+            return problem;
+        } else {
+            return GoalProblem::NO_PROBLEM;
+        }
+    }
+    GoalProblem packages_res = GoalProblem::NO_PROBLEM;
+    for (auto group : group_query) {
+        auto packages = group.get_packages();
+        if (!settings.with_optional) {
+            std::erase_if(packages, [](comps::Package p) { return p.get_type() == comps::PackageType::OPTIONAL; });
+        }
+        auto pkg_settings = GoalJobSettings();
+        pkg_settings.with_provides = false;
+        pkg_settings.with_filenames = false;
+        pkg_settings.nevra_forms.push_back(rpm::Nevra::Form::NAME);
+        for (const auto & pkg : packages) {
+            auto pkg_name = pkg.get_name();
+            auto pkg_condition = pkg.get_condition();
+            if (pkg_condition.empty()) {
+                auto [pkg_problem, pkg_queue] = add_install_to_goal(transaction, pkg_name, pkg_settings);
```

Comment:
I was thinking that to use `add_install_to_goal()` is incorrect but right now I thing it is quite clever because of multilib setting. Only one thing that may be can be improved are reports of missing packages. May be it would be nice to report missing packages from groups by another way or to allow identification of reports generated from user requested job (package install) or as an consequence of other request (installation of package required by a group).

Reply:
I'll add new action `INSTALL_BY_GROUP`.